dojo/tools/api_msdefender/parser.py

```diff
@@ -0,0 +1,125 @@
+                description += "machine Info: osBuild: " + str(machine['osBuild']) + "\n"
+                description += "machine Info: healthStatus: " + str(machine['healthStatus']) + "\n"
+                description += "machine Info: deviceValue: " + str(machine['deviceValue']) + "\n"
+                description += "machine Info: rbacGroupId: " + str(machine['rbacGroupId']) + "\n"
+                description += "machine Info: rbacGroupName: " + str(machine['rbacGroupName']) + "\n"
+                description += "machine Info: riskScore: " + str(machine['riskScore']) + "\n"
+                description += "machine Info: exposureLevel: " + str(machine['exposureLevel']) + "\n"
+                description += "machine Info: isAadJoined: " + str(machine['isAadJoined']) + "\n"
+                description += "machine Info: aadDeviceId: " + str(machine['aadDeviceId']) + "\n"
+                description += "machine Info: defenderAvStatus: " + str(machine['defenderAvStatus']) + "\n"
+                description += "machine Info: onboardingStatus: " + str(machine['onboardingStatus']) + "\n"
+                description += "machine Info: osArchitecture: " + str(machine['osArchitecture']) + "\n"
+                description += "machine Info: managedBy: " + str(machine['managedBy']) + "\n"
+                description += "machine Info: ipAddresses: " + str(machine['ipAddresses']) + "\n"
+                title = str(vulnerability['cveId'])
+                if str(machine['computerDnsName']) != "null":
+                    title = title + "_" + str(machine['computerDnsName'])
+                if str(machine['osPlatform']) != "null":
+                    title = title + "_" + str(machine['osPlatform'])
+                finding = Finding(
+                    title=title + "_" + vulnerability["machineId"],
+                    severity=vulnerability['severity'],
+                    description=description,
+                    static_finding=False,
+                    dynamic_finding=True,
+                    unique_id_from_tool=vulnerability["id"],
+                )
+                if vulnerability['fixingKbId'] is not None:
+                    finding.mitigation = vulnerability['fixingKbId']
+                if vulnerability['cveId'] is not None:
+                    finding.cve = vulnerability['cveId']
+                findings.append(finding)
+                finding.unsaved_endpoints = list()
+                if machine['computerDnsName'] is not None:
+                    finding.unsaved_endpoints.append(Endpoint(host=str(machine['computerDnsName'])))
+                if machine['lastIpAddress'] is not None:
+                    finding.unsaved_endpoints.append(Endpoint(host=str(machine['lastIpAddress'])))
+                if machine['lastExternalIpAddress'] is not None:
+                    finding.unsaved_endpoints.append(Endpoint(host=str(machine['lastExternalIpAddress'])))
+            except IndexError:
```

Comment:
At which moment have you seen `IndexError`? It might be better to avoid raising it than catching it.
Similar to the `KeyError` principle: https://documentation.defectdojo.com/contributing/how-to-write-a-parser/#data-could-be-missing-in-the-source-report

Reply:
While testing, I had one time the issue that a machineid was not present resulting in an Index Error in here: https://github.com/DefectDojo/django-DefectDojo/blob/093b1418cd034d10faca165382c74675e2226277/dojo/tools/api_msdefender/parser.py#L48 
Thus, I also changed the deduplication algorithm to delete the "exception block findings" in the next run as the title is also considered.